### ISSUE TITLE
[FR] Allow users to set the unmarshalling mode as RAISE for BaseConfig

### ISSUE THREAD

**Issue by dragosmc** [ludwig-ai/ludwig#3218]
**Is your feature request related to a problem? Please describe.**
Currently, when loading the user config, Ludwig allows unknown attributes to be set. There is a TODO item in the code at `ludwig/schema/utils.py:161`

```
unknown = INCLUDE  # TODO: Change to RAISE and update descriptions once we want to enforce strict schemas.
```

Since changing this to RAISE by default might impact lots of users I still believe there's value in having this set up as RAISE. That being said it'd be great if users can overwrite this default via an environment variable.

**Describe the use case**
Allow users to enforce the schema on their config.

**Describe the solution you'd like**
The ability to overwrite the default behaviour of INCLDUE

**Describe alternatives you've considered**
I've tried monkeypatching ludwig. Although it's easily doable for a local environment, it becomes harder when sharing the monkeypatch with a team that runs everything in AWS Sagemaker.

**Additional context**
--

**Comment by ksbrar** [comment]
It's a great idea! Wish I had thought of it ;)

**Comment by tgaddair** [comment]
I think that's a great idea @justinxzhao. 

@dragosmc I should also point out that RAISE will become the default behavior in v0.8, so the next release. But we can definitely add this in the meantime.

@ksbrar can you put together a quick PR?

**Comment by dragosmc** [comment]
> I think that's a great idea @dragosmc and @justinxzhao.
> 
> @dragosmc I should also point out that RAISE will become the default behavior in v0.8, so the next release. But we can definitely add this in the meantime.
> 
> @ksbrar can you put together a quick PR?

Sounds good. Thank you all for the prompt change.

**Comment by ksbrar** [comment]
@dragosmc We have just released 0.7.3, https://github.com/ludwig-ai/ludwig/releases/tag/v0.7.3 which contains this FR.

Please use a local environment variable called `LUDWIG_SCHEMA_VALIDATION_POLICY` and set it to either `exclude` (default) or `raise`.
